contracts/Depository.sol

```diff
@@ -226,56 +234,55 @@ contract Depository is IErrorsTokenomics {
 
         // Push newly created bond product into the list of products
         productId = productCounter;
-        mapBondProducts[productId] = Product(uint160(priceLP), uint32(expiry), token, uint96(supply));
+        mapBondProducts[productId] = Product(uint160(priceLP), uint32(vesting), token, uint96(supply));
         // Even if we create a bond product every second, 2^32 - 1 is enough for the next 136 years
         productCounter = uint32(productId + 1);
-        emit CreateProduct(token, productId, supply, priceLP, expiry);
+        emit CreateProduct(token, productId, supply, priceLP, vesting, block.timestamp);
     }
 
     /// @dev Closes bonding products.
     /// @notice This will terminate programs regardless of their vesting time.
     /// @param productIds Set of product Ids.
+    /// @return numClosedProducts Number of closed products.
     /// #if_succeeds {:msg "productCounter not touched"} productCounter == old(productCounter);
-    /// #if_succeeds {:msg "success closed"} forall (uint k in productIds) mapBondProducts[productIds[k]].expiry == 0 && mapBondProducts[productIds[k]].supply == 0;
-    function close(uint256[] memory productIds) external {
+    /// #if_succeeds {:msg "success closed"} forall (uint k in productIds) mapBondProducts[productIds[k]].vesting == 0 && mapBondProducts[productIds[k]].supply == 0;
+    function close(uint256[] memory productIds) external returns (uint256 numClosedProducts) {
```

Comment:
Hm why not return the ids of the closed products?

Reply:
We can do this, however one can be read IDs from events. This implementation is just easier.

Reply:
The thing is, returning the `numClosedProducts` is totally pointless afaict. It tells the "on-chain" caller nothing (as they don't have access to the events). The "off-chain" caller has access to the event in principle but must construct another call to get them.

Reply:
I will restructure.